.github/workflows/source_aws.yml

```diff
@@ -88,39 +100,49 @@ jobs:
       - name: Run sync
         if: startsWith(github.head_ref, 'release-please--branches--main--components')
         run: cloudquery sync test/sanity.yml --log-console
+      - name: Save build cache
+        if:   github.ref == 'refs/heads/main'
+        uses: actions/cache/save@v3
+        with:
+          path: |
+                ~/.cache/go-build
+                ~/Library/Caches/go-build
+                ~\AppData\Local\go-build
+          key:  source-aws-${{ runner.os }}-go-${{ steps.set-up-go.outputs.go-version }}-${{ hashFiles('plugins/source/aws/go.sum') }}
   validate-release:
+    if: startsWith(github.head_ref, 'release-please--branches--main--components') || github.event_name == 'push'
     timeout-minutes: 60
     needs: [resolve-runner]
     runs-on: ${{ needs.resolve-runner.outputs.runner }}
     env:
       CGO_ENABLED: 0
     steps:
       - name: Checkout
-        if: startsWith(github.head_ref, 'release-please--branches--main--components') || github.event_name == 'push'
         uses: actions/checkout@v3
-      - uses: actions/cache@v3
-        if: startsWith(github.head_ref, 'release-please--branches--main--components') || github.event_name == 'push'
-        with:
-          path: |
-            ~/.cache/go-build
-            ~/go/pkg/mod
-          key: ${{ runner.os }}-go-1.19.5-release-cache-${{ hashFiles('plugins/source/aws/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-1.19.5-release-cache-plugins-source-aws
       - name: Set up Go
-        if: startsWith(github.head_ref, 'release-please--branches--main--components') || github.event_name == 'push'
+        id:   set-up-go
         uses: actions/setup-go@v3
         with:
-          go-version-file: plugins/source/aws/go.mod
+          go-version-file:       plugins/source/aws/go.mod
+          cache:                 true
+          cache-dependency-path: plugins/source/aws/go.sum
+      - name: Restore build cache
+        uses: actions/cache/restore@v3
+        with:
+          path:         |
+                        ~/.cache/go-build
+                        ~/Library/Caches/go-build
+                        ~\AppData\Local\go-build
```

Comment:
Since we run only on Linux can we remove the `~/Library/Caches/go-build` and `~\AppData\Local\go-build`? See https://github.com/actions/cache/blob/04f198bf0b2a39f7230a4304bf07747a0bddf146/examples.md#linux-1

Reply:
I used this just to be consistent. These directories don't exist on the runner & it doesn't fail the cache, however, if we ever decide to support non-linux machines it's be easier.

Reply:
removed this for now